### ISSUE TITLE
Error while routing with mode "heavy vehicle"

### ISSUE THREAD

**Issue by danceb** [qgis/qwc2-demo-app#450]
While routing with the mode "heavy vihicle" I am getting the error
```
Uncaught TypeError: this.props.icon is undefined
    render webpack://QWC2App/./qwc2/components/Icon.jsx?:60
```

**Comment by manisandro** [comment]
I suppose this is when computing a route? What reply do you get from valhalla for that route?

**Comment by danceb** [comment]
Yes, this occurs, when the route is calculated. As a result of the query to valhalla I am getting the expeced route result.

```
{"trip":{"locations":[{"type":"break","lat":50.925621,"lon":11.569243,"side_of_street":"right","original_index":0},{"type":"break","lat":50.885984,"lon":11.602049,"side_of_street":"left","original_index":1}],"legs":[{"maneuvers":[{"type":2,"instruction":"Auf Gustav-Eichhorn-Straße Richtung Süden fahren.","verbal_succinct_transition_instruction":"Richtung Süden fahren. Dann Rechts auf Fritz-Reuter-Straße abbiegen.","verbal_pre_transition_instruction":"Auf Gustav-Eichhorn-Straße Richtung Süden fahren. Dann Rechts auf Fritz-Reuter-Straße abbiegen.","verbal_post_transition_instruction":"50 Meter weiter der Route folgen.","street_names":["Gustav-Eichhorn-Straße"],"time":5.426,"length":0.044,"cost":101.354,"begin_shape_index":0,"end_shape_index":4,"verbal_multi_cue":true,"travel_mode":"drive","travel_type":"tractor_trailer"},{"type":10,"instruction":"Rechts auf Fritz-Reuter-Straße abbiegen.","verbal_transition_alert_instruction":"Rechts auf Fritz-Reuter-Straße abbiegen.","verbal_succinct_transition_instruction":"Rechts abbiegen. Dann Links auf Tatzendpromenade abbiegen.","verbal_pre_transition_instruction":"Rechts auf Fritz-Reuter-Straße abbiegen. Dann Links auf Tatzendpromenade abbiegen.","verbal_post_transition_instruction":"80 Meter weiter der Route folgen.","street_names":["Fritz-Reuter-Straße"],"time":16.143,"length":0.082,"cost":86.785,"begin_shape_index":4,"end_shape_index":7,"verbal_multi_cue":true,"travel_mode":"drive","travel_type":"tractor_trailer"},{"type":15,"instruction":"Links auf Tatzendpromenade abbiegen.","verbal_transition_alert_instruction":"Links auf Tatzendpromenade abbiegen.","verbal_succinct_transition_instruction":"Links abbiegen.","verbal_pre_transition_instruction":"Links auf Tatzendpromenade abbiegen.","verbal_post_transition_instruction":"400 Meter weiter der Route folgen.","street_names":["Tatzendpromenade"],"time":54.375,"length":0.431,"cost":175.639,"begin_shape_index":7,"end_shape_index":23,"travel_mode":"drive","travel_type":"tractor_trailer"},{"type":26,"instruction":"In den Kreisverkehr fahren und die 2. Ausfahrt auf Tatzendpromenade nehmen.","verbal_transition_alert_instruction":"In den Kreisverkehr fahren und die zweite Ausfahrt nehmen auf Tatzendpromenade.","verbal_succinct_transition_instruction":"In den Kreisverkehr fahren und die zweite Ausfahrt nehmen.","verbal_pre_transition_instruction":"In den Kreisverkehr fahren und die zweite Ausfahrt nehmen auf Tatzendpromenade.","time":5.064,"length":0.07,"cost":5.816,"begin_shape_index":23,"end_shape_index":29,"roundabout_exit_count":2,"travel_mode":"drive","travel_type":"tractor_trailer"},{"type":27,"instruction":"Aus dem Kreisverkehr auf Tatzendpromenade ausfahren.","verbal_succinct_transition_instruction":"Aus dem Kreisverkehr ausfahren.","verbal_pre_transition_instruction":"Aus dem Kreisverkehr auf Tatzendpromenade ausfahren.","verbal_post_transition_instruction":"3 Kilometer weiter der Route folgen.","street_names":["Tatzendpromenade"],"time":302.384,"length":3.072,"cost":404.432,"begin_shape_index":29,"end_shape_index":207,"travel_mode":"drive","travel_type":"tractor_trailer"},{"type":10,"instruction":"In Richtung Zentrum/Lobeda nach Rechts abbiegen.","verbal_transition_alert_instruction":"In Richtung Zentrum nach Rechts abbiegen.","verbal_succinct_transition_instruction":"In Richtung Zentrum, Lobeda nach Rechts abbiegen.","verbal_pre_transition_instruction":"In Richtung Zentrum, Lobeda nach Rechts abbiegen.","verbal_post_transition_instruction":"400 Meter weiter der Route folgen.","street_names":["Rudolstädter Straße","L 2308"],"time":30.397,"length":0.363,"cost":42.036,"begin_shape_index":207,"end_shape_index":224,"sign":{},"travel_mode":"drive","travel_type":"tractor_trailer"},{"type":15,"instruction":"In Richtung Zentrum/Lobeda nach Links abbiegen.","verbal_transition_alert_instruction":"In Richtung Zentrum nach Links abbiegen.","verbal_succinct_transition_instruction":"In Richtung Zentrum, Lobeda nach Links abbiegen.","verbal_pre_transition_instruction":"In Richtung Zentrum, Lobeda nach Links abbiegen.","verbal_post_transition_instruction":"900 Meter weiter der Route folgen.","street_names":["Lobedaer Straße"],"time":70.373,"length":0.89,"cost":89.703,"begin_shape_index":224,"end_shape_index":261,"sign":{},"travel_mode":"drive","travel_type":"tractor_trailer"},{"type":10,"instruction":"Rechts auf Keßlerstraße abbiegen.","verbal_transition_alert_instruction":"Rechts auf Keßlerstraße abbiegen.","verbal_succinct_transition_instruction":"Rechts abbiegen. Dann Links auf Göschwitzer Straße abbiegen.","verbal_pre_transition_instruction":"Rechts auf Keßlerstraße abbiegen. Dann Links auf Göschwitzer Straße abbiegen.","verbal_post_transition_instruction":"80 Meter weiter der Route folgen.","street_names":["Keßlerstraße"],"time":11.738,"length":0.078,"cost":23.83,"begin_shape_index":261,"end_shape_index":268,"verbal_multi_cue":true,"travel_mode":"drive","travel_type":"tractor_trailer"},{"type":15,"instruction":"Links auf Göschwitzer Straße/L 2308 abbiegen.","verbal_transition_alert_instruction":"Links auf Göschwitzer Straße abbiegen.","verbal_succinct_transition_instruction":"Links abbiegen.","verbal_pre_transition_instruction":"Links auf Göschwitzer Straße, L 2308 abbiegen.","verbal_post_transition_instruction":"einen Kilometer weiter der Route folgen.","street_names":["Göschwitzer Straße","L 2308"],"time":86.378,"length":1.077,"cost":111.531,"begin_shape_index":268,"end_shape_index":318,"travel_mode":"drive","travel_type":"tractor_trailer"},{"type":15,"instruction":"Links auf Prüssingstraße abbiegen.","verbal_transition_alert_instruction":"Links auf Prüssingstraße abbiegen.","verbal_succinct_transition_instruction":"Links abbiegen.","verbal_pre_transition_instruction":"Links auf Prüssingstraße abbiegen.","verbal_post_transition_instruction":"700 Meter weiter der Route folgen.","street_names":["Prüssingstraße"],"time":52.514,"length":0.706,"cost":93.217,"begin_shape_index":318,"end_shape_index":341,"travel_mode":"drive","travel_type":"tractor_trailer"},{"type":6,"instruction":"Ziel befindet sich auf der linken Seite.","verbal_transition_alert_instruction":"Das Ziel befindet sich auf der linken Seite.","verbal_pre_transition_instruction":"Ziel befindet sich auf der linken Seite.","time":0.0,"length":0.0,"cost":0.0,"begin_shape_index":341,"end_shape_index":341,"travel_mode":"drive","travel_type":"tractor_trailer"}],"summary":{"has_time_restrictions":false,"has_toll":false,"has_highway":false,"has_ferry":false,"min_lat":50.885912,"min_lon":11.565836,"max_lat":50.925595,"max_lon":11.600824,"time":634.796,"length":6.816,"cost":1134.346},"shape":"u`gc`B{tcaUn@JbDVxNdChARmCdS}Fr_@uA`OhIExIoAfNwC`r@eUxHRbIzAtVtIlE@pC?rCFhCe@bb@sTfCyAlXuO|G_ElG}C`FgB~EcBxFuAlFoAhC[tC]zLwAvJ}@tL_AbTyAtBOjDWdEAbIb@rFn@tFp@x@JhN~AhJbAfBRtBRjNlArWjClALbXhC|OrBdLjAtKrAf^zDhEXtFCfFQ`BIjB[p@SrCuAfA_AxAwAjA}AlAeCt@kBf@cBHe@dAmF\\sB`@}Bz@gEdAeD`AeCtAgCdAsAfAgApA_A~A]dAEnABrAH`MlArF\\pGGlAEj@Et@IfAQpCq@~DkAbDuAxL}GZQvKqMhCwGlC}I~AcJv@uEpEgNtKqQnHmIhEwDdUwRlSyLjE_AfBa@|INvTbC|[bEXFjVvElTnGvHh@hIBdMBlEHrDn@zAn@`Af@x@f@t@`@fGzCjN`HxPtIt_@pP`FzBnFjCrAhA|GzJ|DtI|@fClFfV|En\\fB~LlJtk@xPv`Ap@hDxCjObDpOtAbGnAxE`@vAt@dCrB~GvBlGnA|CzA|C`GxJtEnFnJ|ItHzF`GnCvLdDtBZvFl@|GDpBC|@Ap@IdDWrBWrDi@r@OvD}@hGuB~F_DdFwDhEeEhDkDlDqE~F{JfIkPhAeDxAmEjDoLhDuRzFob@`TyuBr@aHzBwQnByKl@sDnEcVdDyN`DaNdGkUhCgIlBqGnAaEnLed@nEsTTwAlCiOdB{Kb@_DhBkO`M{kA`Da\\jCwX|BcWf@}Hn@yL`@aM\\kV?cCCmCCuAYoHxImClIeCxS_HpZyL~W_Lx@]x@[vNwFxBy@lKgEvI_ExDwB`C}A~CeDfEqFlR{V~DmFoAiDi@uAo@_CWiAMkAMkAKaBIgCnDee@zCm`@~B{]x@_OvDgw@nA_YJoGBeIW_a@M_d@OwWCmMBqIr@yUHgIRybB@_U?wY@mCGcl@AcGAkK?eXAmVNs`@Nw^?o`@?}ADkPfE`@x@FjBP~LhArE`@vE?dCeCbBmGRaP`@y_@ZuZRuNNmLh@{c@ZmVt@cMtAsMxBeIbC}GvC{F~AwBjFkEhHaE~OsEdUkCzFk@xJkA`GcArFsAzBkAjA{AjBaDjAiCnCuEjE{Dl@k@tL{DfIqB|Fc@hDLpFl@lE|A~EtCnEnEbeAxsA~MfQ~FbH|FpDnDjAhEb@lEAhL_AdL{@lE_@jE_@ps@aGzSaBDwHBaFPkKdAuIjCkEfJmOzLsSvIoO`Tw`@lMkWnVwf@dG{LfEuI|U{g@`O}XpFgGxE_GzLkGtEs@tB]xcA}IzXaChKy@"}],"summary":{"has_time_restrictions":false,"has_toll":false,"has_highway":false,"has_ferry":false,"min_lat":50.885912,"min_lon":11.565836,"max_lat":50.925595,"max_lon":11.600824,"time":634.796,"length":6.816,"cost":1134.346},"status_message":"Found route between points","status":0,"units":"kilometers","language":"de-DE"},"id":"valhalla_directions"}
```

**Comment by manisandro** [comment]
Thanks, should be fixed by 0cb2e9c9c396ad1caaa4542af89ceb6088a1fd63.
